pages/levels/freeplay/index.js

```diff
@@ -148,6 +153,58 @@ export default function FreePlay() {
             }
           })
         );
+      }
+      //windmill:
+      else if (gatherRessources.building === "windmill") {
+        setArray(
+          array.map(tile => {
+            if (tile.id === gatherRessources.tileId) {
+              return {...tile, color: "windmill"};
+            }
+            //left side:
+            else if (
+              (tile.id === gatherRessources.tileId - 11 ||
+                tile.id === gatherRessources.tileId - 1 ||
+                tile.id === gatherRessources.tileId + 9) &&
+              gatherRessources.tileId % 10 !== 0 &&
+              tile.color === "grass" &&
+              tile.dark === false
```

Comment:
You always repeat this 2 conditions. Maybe there is a other solution as to repeat them.